extensions/amp-next-page/1.0/service.js

```diff
@@ -380,33 +385,37 @@ export class NextPageService {
    * Create a container element for the document and insert it into
    * the amp-next-page element
    * @param {!Page} page
-   * @return {!Element}
+   * @return {!Promise<!Element>}
    */
   createDocumentContainerForPage(page) {
     const container = this.win_.document.createElement('div');
     container.classList.add(DOC_CONTAINER_CLASS);
-
-    const shadowRoot = this.win_.document.createElement('div');
-    shadowRoot.classList.add(SHADOW_ROOT_CLASS);
-    container.appendChild(shadowRoot);
+    this.element_.insertBefore(container, this.moreBox_);
 
     // Insert the separator
-    container.appendChild(this.separator_.cloneNode(true));
-
-    // Insert the container
-    this.element_.insertBefore(container, this.moreBox_);
+    const separatorInstance = this.separator_.cloneNode(true);
+    container.appendChild(separatorInstance);
+    // TODO(wassgha): If async/wait was allowed, this would look a lot nicer!
+    return this.maybeRenderSeparatorTemplate_(separatorInstance, page).then(
+      () => {
```

Comment:
If you want to make it a little more legible, you could pull the `then` clause into a dedicated method that takes and returns `container`.

Reply:
I guess I could just name the promise and wait for it afterwards too